src/main/java/kitchenpos/domain/menu/MenuProducts.java

```diff
@@ -0,0 +1,66 @@
+package kitchenpos.domain.menu;
+
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.math.BigDecimal;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static javax.persistence.FetchType.LAZY;
+
+@Embeddable
+public class MenuProducts {
+
+    @OneToMany(fetch = LAZY, mappedBy = "menu")
+    private List<MenuProduct> value;
+
+    public static MenuProducts of(List<MenuProduct> menuProducts) {
+        return new MenuProducts(menuProducts);
+    }
+
+    private MenuProducts(List<MenuProduct> value) {
+        this.value = value;
+    }
+
+    public MenuProducts() {
+    }
+
+    public List<MenuProduct> getValue() {
+        return Collections.unmodifiableList(value);
+    }
+
+    public Price sumOfMenuProductPrice() {
+        BigDecimal sum = BigDecimal.ZERO;
+        for (final MenuProduct menuProduct : value) {
+            final Product product = menuProduct.getProduct();
+            Price multiply = product.multiply(BigDecimal.valueOf(menuProduct.getQuantity()));
+            sum = sum.add(multiply.value);
+        }
```

Comment:
```suggestion
        BigDecimal sum = value.stream()
                .reduce(BigDecimal.ZERO, menuProduct -> {
                    final Product product = menuProduct.getProduct();
                    Price multiply = product.multiply(BigDecimal.valueOf(menuProduct.getQuantity()));
                    sum = sum.add(multiply.value);
                    return sum;
                })
```
이런식으로 표현할 수 있지 않을까요?